src/components/TextInput/v1/TextInput.md

```diff
@@ -0,0 +1,294 @@
+    <TextInput name="example" placeholder="Hint" value="value" isReadOnly />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isReadOnly isDark />
+  </div>
+</div>
+```
+###### Input Icons
+Text inputs have the ability to add a custom icon to the input to give extra content to the input. Text input icons has two available props to customize the input.
+ - `icon` The icon node you want to display
+ - `onIconClick` Click event handler for icon clicks.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" icon={<i className="fab fa-pied-piper" />} onIconClick={(event) => alert("Input Icon Clicked!")} />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" icon={<i className="fab fa-pied-piper" />}  onIconClick={(event) => alert("Input Icon Clicked!")} isDark />
+  </div>
+</div>
+```
+The default valid, invalid and clear icons can be overwritten by passing values to these props of the text input.
+ - `iconClear, iconError, iconSuccess`
+ - `iconClearAccessibilityText`
+
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="Valid" hasBeenValidated iconClear={<i className="fas fa-gavel" />} iconSuccess={<i className="fa fa-thumbs-up" />}  />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="Invalid" errors={["error"]} iconClear={<i className="fas fa-gavel" />} iconError={<i className="fa fa-thumbs-down" />}  isDark />
+  </div>
+</div>
+```
+All of the icon props will accept a few different nodes to create the input icon.
+ - React Component
+ - `<i/>` tag used for icon fonts
+ - `<svg>` tag for inline svg
+ - Plain text
+ 
+Inline SVG icon
+```jsx
+const svg = <svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14"><path d="M9.926 9.105l-2.105-2.105 2.105-2.105-0.82-0.82-2.105 2.105-2.105-2.105-0.82 0.82 2.105 2.105-2.105 2.105 0.82 0.82 2.105-2.105 2.105 2.105zM7 1.176c3.227 0 5.824 2.598 5.824 5.824s-2.598 5.824-5.824 5.824-5.824-2.598-5.824-5.824 2.598-5.824 5.824-5.824z"></path></svg>;
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" icon={svg} />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" icon={svg} isDark />
+  </div>
+</div>
+```
+
+#### Multi line text field
+```jsx
```

Comment:
Should have a sentence above here explaining that you pass `shouldAllowLineBreaks` to enable multi-line